services/sr_asset_assign.py

```diff
@@ -0,0 +1,267 @@
+""" Fetch 3-1-1 SR records from Knack and assign a nearby asset """
+
+import argparse
+import logging
+import os
+import requests
+
+import arrow
+from config.knack import CONFIG, APP_TIMEZONE
+from config.locations import ASSET_CONFIG
+import utils
+import knackpy
+
+APP_ID = os.getenv("KNACK_APP_ID")
+API_KEY = os.getenv("KNACK_API_KEY")
+PGREST_JWT = os.getenv("PGREST_JWT")
+PGREST_ENDPOINT = os.getenv("PGREST_ENDPOINT")
+AGOL_USER = os.getenv("AGOL_USERNAME")
+AGOL_PASS = os.getenv("AGOL_PASSWORD")
+
+
+def create_login_token():
+    """
+    Returns an auth token from AGOL, to hopefully reduce the risk of rate-limiting
+    h/t: https://community.esri.com/t5/arcgis-rest-api-questions/using-python-to-generate-access-token-for-an/td-p/1133618
+    """
+    organizationName = "austin"
+    login_url = f"https://{organizationName}.maps.arcgis.com/sharing/generateToken"
+    params = {
+        "username": AGOL_USER,
+        "password": AGOL_PASS,
+        "expiration": 60 * 4,  # In minutes
+        "f": "json",
+        "referer": f"https://{organizationName}.maps.arcgis.com/",
+    }
+    res = requests.post(url=login_url, data=params)
+
+    return res.json()["token"]
+
+
+def point_in_poly(service_name, layer_id, params):
+    """
+    Check if point is within polygon feature and return attributes of containing
+    feature. Assume input geometry spatial reference is 2277 (NAD83 / Texas Central (ftUS)).
+    docs: http://resources.arcgis.com/en/help/arcgis-rest-api/index.html#//02r3000000p1000000
+    """
+    query_url = f"https://services.arcgis.com/0L95CJ0VTaxqcmED/ArcGIS/rest/services/{service_name}/FeatureServer/{layer_id}/query"
+
+    if "spatialRel" not in params:
+        params["spatialRel"] = "esriSpatialRelIntersects"
+
+    res = requests.get(query_url, params=params)
+    res.raise_for_status()
+
+    return res.json()
+
+
+def asset_filter(field, value):
+    """
+    Provides a filter argument for a searching for matching knack records
+
+    Parameters
+    ----------
+    field : string
+        Field we are querying, must start with field_
+        ex: "field_123"
+    value : string
+        Key value we are searching for.
+
+    Returns
+    -------
+    dict
+        formatted filter argument for Knackpy
+
+    """
+    return {
+        "match": "or",
+        "rules": [{"field": f"{field}", "operator": "is", "value": f"{value}"}],
+    }
+
+
+def get_params(layer_config, point, token):
+    """base params for AGOL query request
+
+    Parameters
+    ----------
+    layer_config : TYPE
+        Description
+
+    Returns
+    -------
+    TYPE
+        Description
+    """
+    params = {
+        "f": "json",
+        "outFields": "*",
+        "where": "1=1",
+        "geomtryType": "esriGeometryPoint",
```

Comment:
I thought this was an error in copying it from the original sr_asset_assign.py, but it's spelled like that there too. And then also included again but spelled correctly? I think this can be removed then?

Reply:
🙈